### PR TITLE
chore: Person ID overrides by default

### PR DESCRIPTION
**Pull request by Twixes**
## Changes

Person-on-events cleanup:
- `person_id_override_properties_joined` becomes the default in code (we can get rid of the `persons-on-events-person-id-override-properties-joined` flag after this)
- `Team.person_on_events_mode` is now consistent with the project setting (which is based on `Team.modifiers`) – previously `Team.person_on_events_mode` didn't know about the project setting (although this only applies to projects where the setting was explicitly changed)
- Got rid of the ancient `person-on-events-enabled` flag

### CODE REVIEW

**Review by tiina303 — APPROVED** [review]
lgtm, but I'm not the best reviewer for this & I didn't really check query .ambr changes

**Review by mariusandra — COMMENTED** [review]
Changes look good, but seem incompatible with some tests, e.g:
image
image
image

File: posthog/session_recordings/queries/test/session_replay_sql.py [hunk 1]
```diff
@@ -146,6 +148,19 @@ def produce_replay_summary(
...
+
+        # It's best to also create a random analytics event, since sessions querying does a JOIN with events in
+        # any person-ID-on-events mode - sessions without an analytics event are excluded
+        _create_event(
+            distinct_id=data["distinct_id"],
+            event="foobarino",
+            properties={"$session_id": data["session_id"]},
+            team_id=team_id,
+        )
+        flush_persons_and_events()
```
Comment (Twixes) [inline posthog/session_recordings/queries/test/session_replay_sql.py:163]: This turned out to be necessary, as the session recordings query works quite differently with `PersonsOnEventsMode.DISABLED` vs. with `PersonsOnEventsMode.PERSON_ID_OVERRIDE_PROPERTIES_JOINED`. CC @PostHog/team-replay for a sanity check

Comment (pauldambra) [reply posthog/session_recordings/queries/test/session_replay_sql.py:163]: yep... we've got the laziest implementation possible right now.

i want to figure out how we can avoid having our own events query so we can just delegate completely

i guess we need to always `select where session_id in (select $session_id from events where...)` or something but am letting future me deal with that 🤣

File: posthog/hogql/hogql.py [hunk 2]
```diff
@@ -22,12 +23,16 @@ def translate_hogql(
     if query == "":
         raise QueryError("Empty query")
 
+    # TRICKY: As `translate_hogql` is only used in legacy queries (not the all-HogQL ones), we must guard against
+    # the PersonsOnEventsMode.PERSON_ID_OVERRIDE_PROPERTIES_JOINED being used here, as that is not supported in legacyz
```
Comment (anirudhpillai) [inline posthog/hogql/hogql.py:27]:
```suggestion
    # the PersonsOnEventsMode.PERSON_ID_OVERRIDE_PROPERTIES_JOINED being used here, as that is not supported in legacy
```

**Review by anirudhpillai — APPROVED** [review]
Someone from replay should cross check impact

**Review by skoob13 — APPROVED** [review]

File: posthog/hogql_queries/insights/test/test_retention_query_runner.py [hunk 3]
```diff
@@ -831,6 +831,7 @@ def test_rolling_retention(self):
         _create_person(team_id=self.team.pk, distinct_ids=["person2"])
         _create_person(team_id=self.team.pk, distinct_ids=["person3"])
         _create_person(team_id=self.team.pk, distinct_ids=["person4"])
+        _create_person(team_id=self.team.pk, distinct_ids=["person5"])
```
Comment (Twixes) [inline posthog/hogql_queries/insights/test/test_retention_query_runner.py:834]: Just a note that this test was off @aspicer

**Pull request merged**